src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -248,5 +251,47 @@ public void Dispose()
                 disposed = true;
             }
         }
+
+        [DllImport("advapi32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool LogonUser(String lpszUsername, String lpszDomain, String lpszPassword,
+       int dwLogonType, int dwLogonProvider, out SafeAccessTokenHandle phToken);
+
+        internal async Task HandleValidateWindowsAccountRequest(
+            ValidateWindowsAccountRequestParams parameters,
+            RequestContext<ValidateWindowsAccountResult> requestContext)
+        {
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                int separator = parameters.Username.IndexOf("\\");
+                string domainName = (separator > -1) ? parameters.Username.Substring(0, separator) : string.Empty;
+                string userName = (separator > -1) ? parameters.Username.Substring(separator + 1, parameters.Username.Length - separator - 1) : string.Empty;
```

Comment:
If the username doesn't contain a `\\` then you should probably return an error that specifically says that. Right now you'll call the method with empty values which is unnecessary and the error returned would probably be misleading. 

Are you validating the username on the client side is in the form `domain\\username` as well? Would be nice to save some time.

Reply:
The users are not allowed to enter the usernames. It is generated by code on the client side and is not editable.  (Third inputbox in the screenshot).

Reply:
Does that code ensure that there is always a valid domain/username?

Regardless I think you should either : 

1. Remove the check and just require/assume that the value passed in is valid
2. Keep the check but add the return error message if it's not valid

I think #2 would be best in case you end up using this elsewhere or start allowing people to enter in their own credentials

Reply:
Also - are you just not going to support users who want to do the migration using a different account than the one they're using ADS with? Seems like a pretty common scenario - especially in corporate environments.

Reply:
It was an ask from the PM to force the username from the connection